app/carbonapi/http_handlers.go

```diff
@@ -193,9 +193,18 @@ func (app *App) renderHandler(w http.ResponseWriter, r *http.Request) {
 	var targetErrs []error
 	size := 0
 	defer func() {
+		//TODO: cleanup RenderDurationPerPointExp
 		if size > 0 {
 			app.prometheusMetrics.RenderDurationPerPointExp.Observe(time.Since(t0).Seconds() * 1000 / float64(size))
 		}
+		//2xx response code is treated as success
+		if toLog.HttpCode > 0 && toLog.HttpCode/100 == 2 {
+			if toLog.TotalMetricCount < int64(app.config.MaxBatchSize) {
```

Comment:
The `MaxBatchSize` may be dropped in quite near future when we optimize to not pre-send finds. This distinction will then lose the point.

Reply:
This can go to comments as TODO

Reply:
@grzkv even then it would remain as a boundary for classifying simple/complex queries.